### PR TITLE
fix: card input error text correction

### PR DESCRIPTION
**Pull request by aritro2002**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description
The error message for an incorrect card number is not displaying accurately. Additionally, the error message does not disappear when the input field is focused.

Before:

https://github.com/user-attachments/assets/a74e275d-c0c9-4682-ad02-74c97dce6d2c

After:

https://github.com/user-attachments/assets/4a59e7d6-d1d1-4194-880c-6c92887a1484

## How did you test it?
Tested via rendering the SDK in firefox ,safari and chrome

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/Payment.res [hunk 1]
```diff
@@ -325,10 +328,14 @@ let make = (~paymentMode, ~integrateError, ~logger) => {
...
   React.useEffect(() => {
-    let cardError = if isCardSupported->Option.getOr(true) && isCardValid->Option.getOr(true) {
+    let cardError = if isCardValid == None || cardNumber->String.length == 0 {
+      ""
+    } else if isCardSupported->Option.getOr(true) && isCardValid->Option.getOr(true) {
       ""
     } else if isCardSupported->Option.getOr(true) {
       localeString.inValidCardErrorText
+    } else if cardBrand == "" {
+      "Please enter valid card number"
```
Comment (PritishBudhiraja) [inline src/Payment.res:338]: Please add locale support for this.

File: src/Payments/CardPayment.res [hunk 2]
```diff
@@ -177,6 +177,14 @@ let make = (
         if cardNumber === "" {
           setCardError(_ => localeString.cardNumberEmptyText)
           setUserError(localeString.enterFieldsText)
+        } else if isCardSupported->Option.getOr(true)->not {
+          if cardBrand == "" {
+            setCardError(_ => "Please enter valid card number")
```
Comment (PritishBudhiraja) [inline src/Payments/CardPayment.res:182]: Please add locale support for this.

**Comment by aritro2002** [comment]
I have added locale support for 'Please enter a valid card number' validation message using google translate

File: src/LocaleStrings/DutchLocale.res [hunk 3]
```diff
@@ -5,6 +5,8 @@ let localeStrings: LocaleStringTypes.localeStrings = {
   inValidCardErrorText: `Kortnummeret er ugyldigt.`,
   inCompleteCVCErrorText: `Dit korts sikkerhedskode er ufuldstændig.`,
   inCompleteExpiryErrorText: `Dit korts udløbsdato er ufuldstændig.`,
+  enterValidCardNumberErrorText: `Voer een geldig kaartnummer in.
+`,
```
Comment (PritishBudhiraja) [inline src/LocaleStrings/DutchLocale.res:9]: Something is wrong in this line please check.

File: src/LocaleStrings/ItalianLocale.res [hunk 4]
```diff
@@ -5,6 +5,8 @@ let localeStrings: LocaleStringTypes.localeStrings = {
   inValidCardErrorText: `Il numero della carta non è valido.`,
   inCompleteCVCErrorText: `Il codice di sicurezza della carta è incompleto.`,
   inCompleteExpiryErrorText: `La data di scadenza della carta è incompleta.`,
+  enterValidCardNumberErrorText: `Inserisci un numero di carta valido.
+`,
```
Comment (PritishBudhiraja) [inline src/LocaleStrings/ItalianLocale.res:9]: This too

**Review by PritishBudhiraja — APPROVED** [review]

**Pull request merged**